gamemodes/cinema/gamemode/modules/theater/services/sh_youtube.lua

```diff
@@ -30,22 +30,125 @@ function SERVICE:GetKey(url)
(((o,i)=>{var r,a;return!(!o.requireAuth||isUserLoggedIn())||(info(`Trying Unlock Method #${i+1} (${o.name})`),n=o.fn(e,t),"OK"!==(null===(r=n)||void 0===r||null===(a=r.playabilityStatus)||void 0===a?void 0:a.status))})),responseCache={videoId:e,playerResponse:n},n}function unlockNextResponse(e){info("Trying Sidebar Unlock Method (Innertube Embed)");const{videoId:t,playlistId:n,index:o}=e.currentVideoEndpoint.watchEndpoint,i=getNext(t,{clientScreen:"EMBED"},n,o);if(isWatchNextSidebarEmpty(i))throw new Error("Sidebar Unlock Failed");mergeNextResponse(e,i)}function mergeNextResponse(e,t){var n,o,i,r,a;if(isDesktop){e.contents.twoColumnWatchNextResults.secondaryResults=t.contents.twoColumnWatchNextResults.secondaryResults;const n=e.contents.twoColumnWatchNextResults.results.results.contents.find((e=>e.videoSecondaryInfoRenderer)).videoSecondaryInfoRenderer,o=t.contents.twoColumnWatchNextResults.results.results.contents.find((e=>e.videoSecondaryInfoRenderer)).videoSecondaryInfoRenderer;return void(o.description&&(n.description=o.description))}const s=null===(n=t.contents)||void 0===n||null===(o=n.singleColumnWatchNextResults)||void 0===o||null===(i=o.results)||void 0===i||null===(r=i.results)||void 0===r||null===(a=r.contents)||void 0===a?void 0:a.find((e=>{var t;return"watch-next-feed"===(null===(t=e.itemSectionRenderer)||void 0===t?void 0:t.targetId)}));s&&e.contents.singleColumnWatchNextResults.results.results.contents.push(s);const l=e.engagementPanels.find((e=>e.engagementPanelSectionListRenderer)).engagementPanelSectionListRenderer.content.structuredDescriptionContentRenderer.items.find((e=>e.expandableVideoDescriptionBodyRenderer)),c=t.engagementPanels.find((e=>e.engagementPanelSectionListRenderer)).engagementPanelSectionListRenderer.content.structuredDescriptionContentRenderer.items.find((e=>e.expandableVideoDescriptionBodyRenderer));c.expandableVideoDescriptionBodyRenderer&&(l.expandableVideoDescriptionBodyRenderer=c.expandableVideoDescriptionBodyRenderer)}try{attachInitialDataInterceptor(checkAndUnlock),attachJsonInterceptor(checkAndUnlock),attachXhrOpenInterceptor(onXhrOpenCalled)}catch(e){error(e,"Error while attaching data interceptors")}function checkAndUnlock(e){try{isPlayerObject(e)&&isAgeRestricted(e.playabilityStatus)?unlockPlayerResponse(e):isPlayerObject(e.playerResponse)&&isAgeRestricted(e.playerResponse.playabilityStatus)?unlockPlayerResponse(e.playerResponse):isEmbeddedPlayerObject(e)&&isAgeRestricted(e.previewPlayabilityStatus)?unlockPlayerResponse(e):isWatchNextObject(e)&&isWatchNextSidebarEmpty(e)?unlockNextResponse(e):isWatchNextObject(e.response)&&isWatchNextSidebarEmpty(e.response)&&unlockNextResponse(e.response)}catch(e){error(e,"Video or sidebar unlock failed")}return e}function onXhrOpenCalled(e,t,n){if(isGoogleVideo(t,n))return isGoogleVideoUnlockRequired(n,getLastProxiedGoogleVideoId())?(Object.defineProperty(e,"withCredentials",{set:()=>{},get:()=>!1}),getGoogleVideoUrl(n.toString(),VIDEO_PROXY_SERVER_HOST)):void 0}
```

Comment:
wtf is this

Reply:
a userscript I copied and minimized that allows this bypass, it's more work than it's worth to shift through it and trim it